### ISSUE TITLE
Server Error: Please check your server logs or contact tech support. When creating new item.

### ISSUE THREAD

**Issue by AnthonyVV** [frappe/erpnext#10699]
Just started getting a "Server Error: Please check your server logs or contact tech support." when creating a new item. If the item code starts with any number, Server Error happens. If the item code stats with a letter, no issue. I'm on ERPNext: v8.11.0 (master) Frappe Framework: v8.10.1 (master).

Failed to load resource: the server responded with a status of 500 (INTERNAL SERVER ERROR)
desk.min.js?ver=1504720287.0:1557 Traceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/app.py", line 57, in application
    response = frappe.handler.handle()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 22, in handle
    data = execute_cmd(cmd)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 53, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 923, in call
    return fn(*args, **newargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/client.py", line 79, in get_value
    return frappe.db.get_value(doctype, filters, fieldname, as_dict=as_dict, debug=debug)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database.py", line 429, in get_value
    order_by, cache=cache)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database.py", line 473, in get_values
    out = self._get_values_from_table(fields, filters, doctype, as_dict, debug, order_by, update)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database.py", line 599, in _get_values_from_table
    conditions, values = self.build_conditions(filters)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database.py", line 392, in build_conditions
    for f in filters:
TypeError: 'int' object is not iterable

**Comment by AnthonyVV** [comment]
Fired up a fresh install. Same issue.

**Comment by OneiricIT** [comment]
I actually had the same issue however in a custom module. I found that when I had the DocType Autoname set to field:[datafieldname] it would occur. Changing it to prompt was an acceptable workaround in the interim. Also started occurring in latest update.
Symptoms were in a new document in that DocType, as soon as I entered a character in the data field a moment later I would receive the error message.
